src/main/java/ru/mail/polis/homework/io/objects/Serializer.java

```diff
@@ -15,41 +20,126 @@
-     * @param animals Список животных для сериализации
+     *
+     * @param animals  Список животных для сериализации
      * @param fileName файл в который "пишем" животных
      */
     public void defaultSerialize(List<Animal> animals, String fileName) {
+        Path dataFile = Paths.get(fileName);
 
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(Files.newOutputStream(dataFile))) {
+            for (Animal animal : animals) {
+                objectOutputStream.writeObject(animal);
+            }
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
     }
 
     /**
      * 1 балл
      * Реализовать простую дисериализацию, с помощью специального потока для дисериализации объектов
+     *
      * @param fileName файл из которого "читаем" животных
      * @return список животных
      */
     public List<Animal> defaultDeserialize(String fileName) {
-        return Collections.emptyList();
+        Path dataFile = Paths.get(fileName);
+
+        if (Files.notExists(dataFile)) {
+            return null;
+        }
+
+        try (ObjectInputStream objectInputStream = new ObjectInputStream(Files.newInputStream(dataFile))) {
+            return (List<Animal>) objectInputStream.readObject();
+        } catch (IOException | ClassNotFoundException ex) {
+            return null;
+        }
     }
 
 
     /**
      * 2 балла
      * Реализовать ручную сериализацию, с помощью высокоровневых потоков
-     * @param animals Список животных для сериализации
+     *
+     * @param animals  Список животных для сериализации
      * @param fileName файл, в который "пишем" животных
      */
     public void customSerialize(List<Animal> animals, String fileName) {
+        Path dataFile = Paths.get(fileName);
+
+        try {
+            if (Files.notExists(dataFile)) {
+                Files.createFile(dataFile);
+            }
+
+            try (DataOutputStream dataOutputStream = new DataOutputStream(new FileOutputStream(fileName))) {
+                for (Animal animal : animals) {
+                    dataOutputStream.writeUTF(animal.getName());
+                    dataOutputStream.writeUTF(animal.getBirthdate());
+                    dataOutputStream.writeUTF(animal.getArea());
+                    dataOutputStream.writeUTF(animal.getType().toString());
+
+                    dataOutputStream.writeUTF(animal.getFather().getName());
+                    dataOutputStream.writeUTF(animal.getFather().getBirthdate());
+                    dataOutputStream.writeUTF(animal.getFather().getType().toString());
+
+                    dataOutputStream.writeUTF(animal.getMother().getName());
+                    dataOutputStream.writeUTF(animal.getMother().getBirthdate());
+                    dataOutputStream.writeUTF(animal.getMother().getType().toString());
+
+                    dataOutputStream.write(animal.getVaccinations().size());
```

Comment:
Writes the specified byte (the low eight bits of the argument b) to the underlying output stream.